src/categorical_algebra/CSets.jl

```diff
@@ -224,48 +227,36 @@ acsets, loose morphisms are usually preferable. For more information about
 limits and colimits in these categories, see [`TightACSetTransformation`](@ref)
 and [`LooseACSetTransformation`](@ref).
 """
-abstract type ACSetTransformation{S<:TypeLevelSchema,Comp,
-                                  Dom<:StructACSet{S},Codom<:StructACSet{S}} end
-# FIXME: The components `Comp` shouldn't be a type parameter in this abstract
-# type but for now it is retained for backwards compatibility.
-
-ACSetTransformation(components, X::StructACSet{S}, Y::StructACSet{S}) where S =
-  ACSetTransformation{S}(components, X, Y)
-ACSetTransformation(X::StructACSet{S}, Y::StructACSet{S}; components...) where S =
-  ACSetTransformation{S}((; components...), X, Y)
-
-function ACSetTransformation{S}(components, X::StructACSet{S}, Y::StructACSet{S}) where S
-  ob_components = filter(∈(objects(S))∘first, pairs(components))
-  type_components = filter(∈(attrtypes(S))∘first, pairs(components))
-  length(ob_components) + length(type_components) == length(components) ||
-    error("Not all names in $(keys(components)) are objects or attribute types")
-  if isempty(type_components)
-    TightACSetTransformation{S}(ob_components, X, Y)
-  else
-    LooseACSetTransformation{S}(ob_components, type_components, X, Y)
-  end
-end
+abstract type ACSetTransformation end
+abstract type TightACSetTransformation <:ACSetTransformation end
+abstract type LooseACSetTransformation <:ACSetTransformation end
```

Comment:
These three top-level types should each have docstrings because they are exported and because most users will interact with these rather than specific concrete subtypes.